os/arch/xtensa/src/esp32/esp32_i2c.c

```diff
@@ -833,6 +814,10 @@ uint32_t esp32_i2c_setclock(FAR struct i2c_dev_s *dev, uint32_t frequency)
 	struct esp32_i2c_priv_s *priv = (struct esp32_i2c_priv_s *)dev;
 	int i2c_num = priv->i2c_num;
 
+	if (ESP32_I2C_FREQUENCY_STD != frequency && ESP32_I2C_FREQUENCY_FAST != frequency) {
```

Comment:
Protect precedence with parenthesis. With proper gcc flags enabled, this would result in compilation error.